src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/IncrementalHash.net46.cs

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    /// <summary>
+    /// Provides support for computing a hash or HMAC value incrementally across several segments.
+    /// </summary>
+    public sealed class IncrementalHash : IDisposable
```

Comment:
But, to continue to have a conversation essentially with myself, there might be some deltas in the details; since desktop's HashAlgorithm does support incremental operations (with some caveats)... so maybe this entirely separate implementation is required.

I'm sure it made sense at the time :).  (Just like I feel like our internal build didn't allow Array.Empty; hence s_empty below)